src/Network/Wai/Middleware/OpenApi.hs

```diff
@@ -0,0 +1,71 @@
+module Network.Wai.Middleware.OpenApi
+  ( Settings (..)
+  , defaultSettings
+  , ValidationErrors (..)
+  , ValidationError
+  , validateRequestBody
+  , validateResponseBody
+  ) where
+
+import Prelude
+
+import Data.ByteString.Builder (toLazyByteString)
+import Data.ByteString.Lazy qualified as BSL
+import Data.IORef (atomicModifyIORef, newIORef, readIORef)
+import Network.Wai (Middleware, Request, Response)
+import Network.Wai qualified as Wai
+import Network.Wai.Middleware.OpenApi.Schema
+import Network.Wai.Middleware.OpenApi.Settings
+import Network.Wai.Middleware.OpenApi.Validate
+import Network.Wai.Middleware.OpenApi.ValidationError
+
+validateRequestBody :: Settings -> Middleware
+validateRequestBody settings app request respond =
+  validate
+    ValidateOptions
+      { settings
+      , request
+      , schema = lookupRequestSchema settings.spec settings.pathMap request
+      , getBody = getRequestBody request
+      , respondActual = \case
+          Nothing -> app request respond
+          Just newReq -> app newReq respond
+      , respondErrors = respond . clientErrorResponse
```

Comment:
I wonder if we should support the client configuring how the error response is handled?

Reply:
> how the error response is handled?

Not sure what you mean by this.

Reply:
I just mean a user of this library may not want to return a json error

Reply:
The only case I can think of where that would be useful, is if the app already has a schema defined for 400 responses and they want to make sure our own errors fit it. Can you think of any other reasons our current behavior wouldn't be appropriate?

If not, I think I'm comfortable (for now) saying that such an edge-case should be handled by dropping down to `validate`; then they can set `respondErrors` to do whatever they like. WDYT?

Reply:
> The only case I can think of where that would be useful, is if the app already has a schema defined for 400 responses and they want to make sure our own errors fit it. Can you think of any other reasons our current behavior wouldn't be appropriate?

By schema do you mean one for the json? There's that, but it also seems reasonable to send back an html response too.

>If not, I think I'm comfortable (for now) saying that such an edge-case should be handled by dropping down to validate; then they can set respondErrors to do whatever they like. We could even ship some defaultValidateOptions values for them to build on. WDYT?

I don't think it's worth worrying about too much especially if there's a way that someone *could* do it.

Reply:
> but it also seems reasonable to send back an html response too

Well, my thinking on that is that this middleware only operates when the request/response is `application/json`. So the only time it would be reasonable to send back HTML is if someone sent an (invalid) request at `content-type: application/json`, but was looking for a response with `accept: text/html`. That seems unlikely to me.